kotlin/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/arcgisruntime/sample/filterbydefinitionexpressionordisplayfilter/MainActivity.kt

```diff
@@ -0,0 +1,178 @@
+            operationalLayers.add(featureLayer)
+            addDoneLoadingListener {
+                if (loadStatus != LoadStatus.LOADED) {
+                    Log.e(TAG, "Failed to load the feature layer map")
+                    Toast.makeText(this@MainActivity, "Failed to load the feature layer map", Toast.LENGTH_SHORT).show()
+                }
+            }
+        }
+
+        mapView.apply {
+            map = topographicMap
+            val viewpoint = Viewpoint(Point(-122.4401, 37.7722, SpatialReferences.getWgs84()), 100000.0)
+            setViewpoint(viewpoint)
+            // run countFeatures() when MapView is finished moving
+            addNavigationChangedListener {
+                if (!isNavigating)
+                    countFeatures(featureLayer)
+            }
+        }
+
+        expressionButton.setOnClickListener {
+            featureLayer.apply {
+                // Reset the display filter definition
+                displayFilterDefinition = null
+                // Set the definition expression to show specific features only
+                definitionExpression = "req_Type = 'Tree Maintenance or Damage'"
+            }
+            countFeatures(featureLayer)
+        }
+
+        filterButton.setOnClickListener {
+            featureLayer.apply {
+                // Disable the feature layer definition expression
+                definitionExpression = ""
+                // create a display filter and display filter definition
+                // req_type here is one of the published fields
+                val damagedTrees = DisplayFilter("Damaged Trees", "req_type LIKE '%Tree Maintenance%'")
+                // Set the display filter definition on the layer
+                displayFilterDefinition = ManualDisplayFilterDefinition(damagedTrees, listOf(damagedTrees))
+            }
+            countFeatures(featureLayer)
+        }
+
+        resetButton.setOnClickListener {
+            featureLayer.apply {
+                // Disable the feature layer definition expression
+                definitionExpression = ""
+                // Reset the display filter definition
+                displayFilterDefinition = null
+            }
+            countFeatures(featureLayer)
+        }
+    }
+
+    /**
+     * Retrieves the totalFeatureCount from the [featureLayer]'s current
+     * featureTable in the [mapView]'s extent
+     */
+    private fun countFeatures(featureLayer: FeatureLayer) {
+        // Get the extent of the current viewpoint, return if no valid extent.
+        val extent = mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).targetGeometry.extent ?: return
+
+        // Update the UI with the count of features in the extent
+        val queryParameters = QueryParameters().apply {
+            geometry = extent
+        }
+
+        val featureCountFuture = featureLayer.featureTable.queryFeatureCountAsync(queryParameters)
+        featureLayer.featureTable
```

Comment:
Think this line may have been left in accidently?

Reply:
Good catch, had this in while I was testing something